theano/compile/builders.py

```diff
@@ -144,25 +150,59 @@ class OpFromGraph(gof.Op):
 
     .. code-block:: python
 
-        from thenao import funciton, OpFromGraph, tensor, grad
+        from theano import function, OpFromGraph, tensor, grad
         x, y, z = tensor.scalars('xyz')
         e = x + y * z
         def rescale_dy(inps, grads):
             x, y, z = inps
-            g = grads
+            g, = grads
             return z*2
         op = OpFromGraph(
-            [x, y, z], [e], grad_overrides=[None, rescale_dy, None])
+            [x, y, z], [e], grad_overrides=[Ellipsis, rescale_dy, Ellipsis]
         e2 = op(x, y, z)
         dx, dy, dz = grad(e2, [x, y, z])
         fn = function([x, y, z], [dx, dy, dz])
-        # the graident wrt y is now doubled
+        # the gradient wrt y is now doubled
         fn(2., 3., 4.) # [1., 8., 3.]
 
     """
-    def __init__(self, inputs, outputs, inline=False, grad_overrides=None, rop_overrides=None, name=None, **kwargs):
+    ofg_null_t = NullType(why_null='ofg_overridden')
+    ofg_discon_t = DisconnectedType()
+
+    @staticmethod
+    def _filter_grad_var(grad, inp):
+        # Returns (filtered_var, overrider_var)
+        # Args:
+        #     grad: gradient Variable
+        #     inp: the corresponding input of gradient Variable
+        #
+        # Some Variable types cannot be used directly as OfG output such as
+        # NullType, or DisconnectedType.
+        #
+        # However a grad() call could return these types
+        #
+        # Since we always use an OfG instance as self._grad_op, the current
+        # workaround is to "remember" the special cases of the gradient and
+        # replace them after self._grad_op is called.
+        #
+        # This helper function changes invalid types into a filtered_type,
+        # and provides a overrider_type to be replaced at grad() call
+        #
+        # For now, this converts NullType or DisconnectedType into zeros_like.
+        # other types are unmodified with overrider_type -> None
+        if isinstance(grad.type, (NullType, DisconnectedType)):
+            return inp.zeros_like(), grad.type
+        else:
+            return grad, None
+
+    def __init__(
+            self, inputs, outputs,
+            inline=False,
+            grad_overrides=Ellipsis, rop_overrides=Ellipsis,
+            name=None, **kwargs
+    ):
         if not isinstance(outputs, list):
-            raise TypeError('outputs must be list', outputs)
+            raise TypeError('outputs must be list, got %s' % outputs, outputs)
```

Comment:
remove ", outputs". This duplicate the information in the error message. Your change make it more clear which is good.